src/lib.rs

```diff
@@ -317,13 +327,43 @@ impl RccJIT {
     }
     /// Get compiled static data, if this data doesn't exit then `None` is returned, otherwise it's andress and size returned.
     pub fn get_compiled_data(&mut self, name: &str) -> Option<(*mut u8, usize)> {
+        use cranelift_module::FuncOrDataId;
+
         let name = self.module.get_name(name);
         if let Some(FuncOrDataId::Data(id)) = name {
             Some(self.module.get_finalized_data(id))
         } else {
             None
         }
     }
+    /// Given a module, run the `main` function.
+    ///
+    /// This automatically calls `self.finalize()`.
+    /// If `main()` does not exist in the module, returns `None`; otherwise returns the exit code.
+    ///
+    /// # Safety
+    /// This function runs arbitrary C code.
+    /// It can segfault, access out-of-bounds memory, cause data races, or do anything else C can do.
+    pub unsafe fn run_main(&mut self) -> Option<i32> {
+        self.finalize();
+        let main = self.get_compiled_function("main")?;
+        let args = std::env::args().skip(1);
+        let argc = args.len() as i32;
+        let vec_args = args
+            .map(|string| std::ffi::CString::new(string).unwrap())
+            .collect::<Vec<_>>();
+        // CString should be alive if we want to pass it's pointer to another function properly, otherwise this may lead to memory leak or UB.
```

Comment:
```suggestion
        // CString should be alive if we want to pass its pointer to another function properly, otherwise this may lead to memory leak or UB.
```